rs/tests/nns/rent_subnet_test.rs

```diff
@@ -0,0 +1,423 @@
+
+    static ref SUBNET_USER_SENDER: Sender = Sender::from_keypair(&*SUBNET_USER_KEYPAIR);
+
+    static ref SUBNET_USER_PRINCIPAL_ID: PrincipalId = SUBNET_USER_SENDER.get_principal_id();
+}
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(setup)
+        .add_test(systest!(test))
+        .execute_from_args()?;
+    Ok(())
+}
+
+pub fn setup(env: TestEnv) {
+    let mut ic = InternetComputer::new();
+
+    // This hack is needed so that we can install a (mock) Exchange Rate
+    // canister at its usual canister ID. This hack is copied from
+    // rs/tests/testnets/src_testing.rs. What this does is ensure that there is
+    // a system subnet that is assigned a canister ID range containing the usual
+    // Exchange Rate canister ID (uf6dk-hyaaa-aaaaq-qaaaq-cai).
+    for _ in 0..32 {
+        ic = ic.add_subnet(Subnet::new(SubnetType::Application).add_nodes(1));
+    }
+    ic = ic.add_subnet(
+        Subnet::new(SubnetType::System)
+            .with_features(SubnetFeatures {
+                http_requests:true,
+                ..SubnetFeatures::default()
+            })
+            .add_nodes(1),
+    );
+    ic = ic.add_subnet(
+        Subnet::fast(
+            SubnetType::System,
+            1, // Node count.
+        )
+        .with_dkg_interval_length(Height::from(DKG_INTERVAL_LENGTH))
+    );
+
+    ic
+        .with_unassigned_nodes(1)
+        .setup_and_start(&env)
+        .expect("failed to setup IC under test");
+
+    env.topology_snapshot().subnets().for_each(|subnet| {
+        subnet
+            .nodes()
+            .for_each(|node| node.await_status_is_healthy().unwrap())
+    });
+    env.topology_snapshot()
+        .unassigned_nodes()
+        .for_each(|node| node.await_can_login_as_admin_via_ssh().unwrap());
+}
+
+pub fn test(env: TestEnv) {
+    let log = &env.logger();
+
+    // [Phase I] Prepare NNS
+    install_nns_canisters(&env);
+    let topology_snapshot = &env.topology_snapshot();
+    let an_nns_subnet_node = topology_snapshot
+        .root_subnet()
+        .nodes()
+        .next()
+        .unwrap();
+
+    // [Phase II] Execute and validate the testnet changes
+
+
+    block_on(async move {
+        let runtime = new_node_runtime(&an_nns_subnet_node);
+
+        let registry_canister = RegistryCanister::new_with_query_timeout(
+            vec![an_nns_subnet_node.get_public_url()],
+            Duration::from_secs(10),
+        );
+        let original_subnets = get_subnet_list_from_registry(&registry_canister)
+            .await
+            // Convert to HashSet
+            .into_iter()
+            .collect::<HashSet<SubnetId>>();
+        // Not sure why the NNS subnet is not in original_subnets...
+        assert!(!original_subnets.is_empty(), "registry contains no subnets");
+
+        subnet_user_sends_icp_to_the_subnet_rental_canister(&runtime).await;
+
+        execute_subnet_rental_request(&an_nns_subnet_node, *SUBNET_USER_PRINCIPAL_ID).await;
```

Comment:
The test seems to get past this point, so it seems that it is possible to call the subnet rental canister from governance...